etcdserver/etcdserverpb/raft_internal.proto

```diff
@@ -10,10 +10,17 @@ option (gogoproto.sizer_all) = true;
 option (gogoproto.unmarshaler_all) = true;
 option (gogoproto.goproto_getters_all) = false;
 
+message RequestHeader {
+  uint64 ID = 1;
+  // username is a username that is associated with an auth token of gRPC connection
+  string username = 2;
+}
+
 // An InternalRaftRequest is the union of all requests which can be
 // sent via raft.
 message InternalRaftRequest {
-  uint64 ID = 1;
+  RequestHeader header = 1;
```

Comment:
Do you mean we need to keep both of the existing ID and a new ID in the RequestHeader?